actor/builtin/miner/miner.go

```diff
@@ -660,12 +660,23 @@ func (ma *Actor) SubmitPoSt(ctx exec.VMContext, proof []byte) (uint8, error) {
 		postProof := proofs.PoStProof{}
 		copy(postProof[:], proof)
 
+		// See comment above, in CommitSector.
+		//
+		// It is undefined behavior for a miner in "Live" mode to verify a proof
+		// created by a miner in "ProofsTest" mode (and vice-versa).
+		//
+		sectorStoreType := proofs.Live
```

Comment:
would be nicer to pass this in as a parameter to the vm and surface in the context. no action required, just sayin.

Reply:
> would be nicer to pass this in as a parameter to the vm and surface in the context. no action required, just sayin.

The solution that you see is a shameful hack which I am responsible for. I think that moving the environment variable checking to some other codebase is perhaps more aesthetically pleasure, but is still a hack.

We need some mechanism for:

1) Configuring a node such that it uses small or large sectors
2) Prevent the node from being reconfigured to use small sectors after it has been started with a large sector configuration (and vice-versa)
3) Ensuring that the node configured to use small sectors can't join a cluster of nodes using large sectors

The environment variable thing is hacky because it allows a node to come online with small sectors enabled (because the `FIL_USE_SMALL_SECTORS` environment variable was set to `true`), shut down, and be restarted with large sectors enabled.

I am hoping that the "multiple sector sizes" work that @porcuquine (and friends) is doing will elevate the distinction between sector sizes to the protocol, which would allow me to remove this check entirely. That has yet to land, so I'd like to keep this big, gaping wound-of-a-reminder as it is today.